provider/tasks.py

```diff
@@ -126,6 +127,15 @@ def create_submission(task, submission_key, course_key, submission_api_url, matc
     submission.tokens = tokens
     submission.indexes_json = json_indexes
 
+    watermarks = tokenizer.watermarker(submission_text)
+    copied = False
+    if watermarks != None:
+        for i in watermarks:
+            if i != submission.student.id:
+                copied = True
+    submission.watermark = watermarks
```

Comment:
This loop does not do anything since the `copied` variable is not used anywhere. Assigning the `watermarks` list into the `submission.watermark` TextField/CharField probably crashes due to incompatible types.